Makefile.local

```diff
@@ -1,24 +1,145 @@
-# This is the local Makefile for the projectcalico/api repository, with targets
-# specific to github.com/projectcalico/api. This is opposed to Makefile, which
+# This is the local Makefile for the tigera/api repository, with targets
+# specific to github.com/tigera/api. This is opposed to Makefile, which
 # is mirrored from github.com/projectcalico/calico/api.
 
+SHELL = /bin/bash
+
+# Local variables
+
+# The revision of the local repository; this is used as a marker for where/when
+# the PR was generated
+LOCAL_REPOSITORY_REVISION ?= $(shell git describe --tags)
+
+# We assume our git branch will match the upstream
+# git branch we're pulling API updates from, but
+# we can override this if we ever need to.
+GIT_BRANCH := $(shell git branch --show-current)
```

Comment:
This could be a tag / commit too if we wanted, no? It would also be good to name it so that it indicates this is referring to a git reference in the Calico repo, not in the api repo.

e.g., if we wanted to pull in _exactly_ `v3.18.4`

maybe this should be `CALICO_GIT_REF`? Something like that?

Reply:
In the concept of the workflow, having this as any arbitrary commit-ish would probably be reasonable. It falls apart a bit in the workflow since:

1. We need to know a branch name to create a PR
2. We need to be on a branch to get the branch name (e.g. the detached head you get with `git clone -b <tagname>` will not work).

So fundamentally we need to know `GIT_BRANCH` (the API repo branch) and `CALICO_GIT_REF` as an optionally-different ref in the upstream repository (e.g. a tag name).

I've changed the logic to get `GIT_BRANCH` as usual, and then to also set `CALICO_GIT_REF` to the value of `GIT_BRANCH` if `CALICO_GIT_REF` isn't set already. This should allow us to override the ref from the upstream repository if we want, but preserve the current behavior by default.